awspricing/__init__.py

```diff
@@ -28,69 +23,47 @@ def _fetch_offers():
     if offers is not None:
         return offers
 
-    resp = session.get(OFFER_BASE_URL + OFFER_INDEX_ENDPOINT)
-    resp.raise_for_status()
-    offers = resp.json()['offers']
-
     maybe_write_to_cache(cache_key, offers)
     return offers
 
 
-def _fetch_available_versions(offers, offer_name):
-    cache_key = 'offers_version_{}'.format(offer_name)
-    offers_version = maybe_read_from_cache(cache_key)
-    if offers_version is not None:
-        return offers_version
+def _get_services():
+    global service_list
+    if not service_list:
+        service_list = all_services_names()
+    return service_list
 
-    offers_version_endpoint = offers[offer_name]['versionIndexUrl']
-    resp = session.get(OFFER_BASE_URL + offers_version_endpoint)
-    resp.raise_for_status()
-    offers_version = resp.json()['versions']
 
-    maybe_write_to_cache(cache_key, offers_version)
-    return offers_version
-
-
-def _get_offers():
-    global _OFFERS
-    if not _OFFERS:
-        _OFFERS = _fetch_offers()
-    return _OFFERS
-
-
-def _fetch_offer(offer_name, version='current'):
-    offers = _get_offers()
-    if offer_name not in offers:
-        raise ValueError('Unknown offer name: {}'.format(offer_name))
+def _fetch_offer(offer_name, version=None):
+    services = _get_services()
+    if offer_name not in services:
+        raise ValueError('Unknown offer name, no corresponding AWS Service: {}'.format(offer_name))
+    if not version:
+        datetime.datetime.utcnow().strftime(TIME_FORMAT)
```

Comment:
Should this be prepended with `version = `?

Reply:
yep, fixed